src/kernel/filesystem/vfs.zig

```diff
@@ -224,12 +224,28 @@ pub const DirNode = struct {
     pub fn close(self: *const DirNode) void {
         var fs = self.fs;
         var node = self;
-        if (self.mount) |mnt| {
-            fs = mnt.fs;
-            node = mnt;
+
+        // TODO: Use @fieldParentPtr() once implemented for unions
+        const cast_node = @ptrCast(*const Node, node);
+        // Can't close the root node
```

Comment:
Why not?

Reply:
Because the root node is never 'opened' thought the `open` function, it just gets returned by `getRootNode` so can't be closed by `close`.

Reply:
True, we want to keep the root around at all times. I think this has exposed a possible design or mental representation issue with the VFS: we want to let a process open any node (including the root) and close it when they are done since we'd want to release the file handle etc., but if this then called the close function on the node it would close it for all other processes too, which we wouldn't want. Perhaps we need to have a think about what closing *actually* means.

Reply:
We could have it so that the underlying filesystem handle the closing of the root node so if it isn't 'opened', the the filesystem will ignore the request or do what it need to do with closing the root node if it is open.